sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/ConfigurationUtils.java

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.util;
+
+import com.azure.core.util.ConfigurationPropertyBuilder;
+import com.azure.core.util.CoreUtils;
+
+import java.time.Duration;
+import java.util.function.Function;
+
+public class ConfigurationUtils {
+
+    private static final Function<String, Boolean> BOOLEAN_CONVERTER = (value) -> Boolean.parseBoolean(value);
+    private static final Function<String, Duration> DURATION_CONVERTER = (value) -> {
```

Comment:
Hmm, will we want to support timespan durations as well?

Reply:
I think eventually yes, but I didn't see a need for `TimeSpan` configuration properties anywhere in `azure-core` and `appconfiguration`.  Nothing stops some client that needs it to define it from doing so using `ConfigurationPropertyBuilder` directly, so I'd rather postpone adding it until we know it's a common need.